### DISCUSSION TITLE
Hello. How create own library for ev3 micropython 2.0 in vscode?

### DISCUSSION THREAD

**Discussion by IslamSalam** [micropython/micropython#15737] [Using MicroPython] [answered]
I created my own simple module (library) (3rd picture), uploaded it to Pypi and installed it in vscode. When I run the program for lego EV3 (1nd picture) by pressing f5 it says ImportError: no module named, and in another window when I run it simply as python by pressing the button on the right run python file everything works (2st picture), the module import goes well. To be able to program lego ev3 on python, the Lego ev3 micropython and ev3dev-browser extensions were installed. I added the path to the module to path, it didn't help either.
![1](https://github.com/user-attachments/assets/66613e30-afdc-40eb-8e7e-0dfbe8587100)
![2](https://github.com/user-attachments/assets/ede9aed7-6b09-4b04-a6c6-fc704d1ffe4f)
![3](https://github.com/user-attachments/assets/75688667-dd77-4d85-ba31-e03b38dcffc0)
Thanks!!!

**Comment by Josverl** [comment, marked as answer]
Hi , 
it may be better if next time you do not upload screenshots , but your code.

> You can use [GitHub Flavoured Markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github) to format your posts. In particular use ```py ... ``` around your code snippets to get syntax highlighting, see the section below.
> https://github.com/orgs/micropython/discussions/9111 

Micropython does **not** use PyPI for package distribution, In stead op `pip` there is `mip`, documeted on [micropython docs](https://docs.micropython.org/en/latest/reference/packages.html).

During development its much simpler to copy the files from your host to the MCU , 
or use mpremote to mount your your ( in reverse as it were)
for instance by using `mpremote mount . exec "import local_script"`

I dont have an EV3 , so there may be other options as well.

**Reply by IslamSalam** [reply]
Thanks!!!
